typescript/packages/plugins/coingecko/src/tools.ts

```diff
@@ -0,0 +1,58 @@
+import type { DeferredTool } from "@goat-sdk/core";
+import type { z } from "zod";
+
+import {
+    getTrendingCoinsParametersSchema,
+    getCoinPriceParametersSchema,
+} from "./parameters";
+
+// Define methods to interact with CoinGecko API
+async function fetchTrendingCoins() {
+    const response = await fetch('https://api.coingecko.com/api/v3/search/trending');
+    if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+    }
+    return await response.json();
+}
+
+async function fetchCoinPrice(coinId: string, vsCurrency: string) {
+    const params = new URLSearchParams({
+        ids: coinId,
+        vs_currencies: vsCurrency,
+    });
+    const response = await fetch(`https://api.coingecko.com/api/v3/simple/price?${params.toString()}`);
+    if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+    }
+    return await response.json();
+}
+
+export function getTools(credentials: {
+	key: string;
+}): DeferredTool<any>[] {
+    const tools: DeferredTool<any>[] = [];
+
+    const getTrendingCoinsTool: DeferredTool<any> = {
+        name: 'get_trending_coins',
+        description: 'This tool fetches the list of trending coins from CoinGecko',
```

Comment:
Check out the other plugins, use this `{{tool}}` 

Reason explained in the step 2 of the plugin tutorial https://ohmygoat.dev/plugins